### ISSUE TITLE
GATK BuildBamIndex creates index file in the current directory

### ISSUE THREAD

**Issue by regnveig** [broadinstitute/gatk#8017]
Dear colleagues,

I discovered a thing which seems invalid. `BuildBamIndex` creates index file in the current directory, not in the directory which contains bam file. I'm absolutely sure that it behaved as I expected in previous versions. 

Example:

```bash
source ~/miniconda3/bin/activate gatk
cd /some/path
gatk BuildBamIndex -I /path/to/bam/my_file.bam
```

Expected output:

```bash
/path/to/bam/my_file.bai
```

Observed:

```bash
/some/path/my_file.bai
```

**OS:** Ubuntu 22.04
**GATK version:** 4.2.6.1
**Environment:** miniconda3

Could you check that, please?

**Comment by cmnbroad** [comment]
Yes, this is probably a regression introduced in Picard https://github.com/broadinstitute/picard/pull/1787. Im transferring this to the Picard repo.
